### PR TITLE
update 29 formula

### LINKED ISSUE #595: Emacs 29.1 has been released
Please support official 29.1 release. Thanks!

### PR DESCRIPTION
**Pull request by dustinfarris**
Addressing @d12frosted comments in #593 to update 29 target and documentation.

Also includes ahead-of-time native compilation flag.  I added this as a separate flag from regular native compilation — lmk if you'd prefer a different approach.

Closes #595

### CODE REVIEW

**Comment by dustinfarris** [comment]
@d12frosted I am able to install this with `./build 29` but when I try to install from hombrew i get an error:

```
brew tap dustinfarris/homebrew-emacs-plus ~/code/homebrew-emacs-plus
brew install emacs-plus --with-modern-sexy-v2-icon --with-dbus --with-native-comp-aot --with-no-frame-refocus

==> Fetching dustinfarris/emacs-plus/emacs-plus@29
==> Downloading https://ftp.gnu.org/gnu/emacs/emacs-29.1.tar.xz
################################################################################################################################################################################################ 100.0%
==> Installing emacs-plus@29 from dustinfarris/emacs-plus
==> Patching
Error: An exception occurred within a child process:
  Errno::ENOENT: No such file or directory @ rb_sysopen - /Users/a80291516/patches/emacs-29/no-frame-refocus-cocoa.patch
```

I was not getting this error when installing in this way on https://github.com/d12frosted/homebrew-emacs-plus/pull/593/commits/aff45bb255476363e4a05f9aca9234ae760bf165

**Comment by d12frosted** [comment]
> brew tap dustinfarris/homebrew-emacs-plus ~/code/homebrew-emacs-plus

Never tested the 'local_patch' helper with custom path for tap. Try tapping without custom path :)

File: Formula/emacs-plus@29.rb [hunk 1]
```diff
@@ -68,12 +75,10 @@ class EmacsPlusAT29 < EmacsBase
...
     end
   end
-
-  #
-  # URL
-  #
-
-  url "https://github.com/emacs-mirror/emacs.git", :branch => "emacs-29"
+  
+  if build.with? "native-comp" and build.with? "native-comp-aot"
```
Comment (d12frosted) [inline Formula/emacs-plus@29.rb:79]: ideally I would love to introduce extra option separately from 29.1 release 

p.s. what is the difference?

**Review by d12frosted — APPROVED** [review]
👏 thank you very much! that's exactly what needs to be done

if possible, extract aot as a separate PR 🙏 would love to keep commits clean
for you it will be extra contribution ;)

**Comment by dustinfarris** [comment]
> > brew tap dustinfarris/homebrew-emacs-plus ~/code/homebrew-emacs-plus
> 
> Never tested the 'local_patch' helper with custom path for tap. Try tapping without custom path :)

This was because i was tapping (arbitrarily) to `dustinfarris/homebrew-emacs-plus` but the urlresolver is [hard-coded to expect d12frosted](https://github.com/d12frosted/homebrew-emacs-plus/blob/63f242743b45dc9e31d78b0f55e1028da50ad820/Library/UrlResolver.rb#L1).

Once I resolved that I'm able to tap locally and install successfully.

File: Formula/emacs-plus@29.rb (hunk 1 above)
Comment (dustinfarris) [reply Formula/emacs-plus@29.rb:79]: I split the commit to 2 commits.  Do you still want a separate PR for the ahead-of-time commit?

Comment (dustinfarris) [reply Formula/emacs-plus@29.rb:79]: > what is the difference?

all core elisp gets compiled immediately instead of just-in-time during usage.

**Comment by d12frosted** [comment]
Thank you! I cherry picked your first two commits. Please open a separate PR for AOT. And don't forget to add this option to documentation 🙏

**Pull request closed without merging**